pyat/at/matching/globalfit.py

```diff
@@ -4,91 +4,105 @@
                          increment=True)
-        datap = func(ring, dp)
+        datap = func(ring, dp, **kwargs)
         set_value_refpts(ring, refpts, attname, -2*delta, index=index,
                          increment=True)
-        datan = func(ring, dp)
+        datan = func(ring, dp, **kwargs)
         set_value_refpts(ring, refpts, attname, delta, index=index,
                          increment=True)
         data = numpy.subtract(datap, datan)/(2*delta)
         return data
 
     def _fit(ring, index, func, refpts1, refpts2, newval, J,
-             dp: Optional[float] = 0):
-        val = func(ring, dp)
+             dp: Optional[float] = 0, **kwargs):
+        val = func(ring, dp, **kwargs)
         dk = numpy.linalg.solve(J, numpy.subtract(newval, val))
         set_value_refpts(ring, refpts1, 'PolynomB', dk[0], index=index,
                          increment=True)
         set_value_refpts(ring, refpts2, 'PolynomB', dk[1], index=index,
                          increment=True)
-        val = func(ring, dp)
+        val = func(ring, dp, **kwargs)
         sumsq = numpy.sum(numpy.square(numpy.subtract(val, newval)))
         return sumsq
 
     delta = 1e-6 * 10 ** index
-    dq1 = _get_resp(ring, index, func, refpts1, 'PolynomB', delta, dp)
-    dq2 = _get_resp(ring, index, func, refpts2, 'PolynomB', delta, dp)
+    dq1 = _get_resp(ring, index, func, refpts1, 'PolynomB',
+                    delta, dp, **kwargs)
+    dq2 = _get_resp(ring, index, func, refpts2, 'PolynomB',
+                    delta, dp, **kwargs)
     J = [[dq1[0], dq2[0]], [dq1[1], dq2[1]]]
 
     n = 0
     sumsq = tol+1
-    print('Initial value', func(ring, dp))
+    print('Initial value', func(ring, dp, **kwargs))
     while sumsq > tol and n < niter:
-        sumsq = _fit(ring, index, func, refpts1, refpts2, newval, J, dp=dp)
+        sumsq = _fit(ring, index, func, refpts1, refpts2, newval,
+                     J, dp=dp, **kwargs)
         print('iter#', n, 'Res.', sumsq)
         n += 1
-    print('Final value', func(ring, dp), '\n')
+    print('Final value', func(ring, dp, **kwargs), '\n')
     return
 
 
 def fit_tune(ring: Lattice, refpts1: Refpts, refpts2: Refpts, newval,
              tol: float = 1.0e-12,
-             dp: Optional[float] = 0, niter: int = 3) -> None:
+             dp: Optional[float] = 0, niter: int = 3,
+             **kwargs) -> None:
     """Fits the tunes using 2 families
 
     Args:
         ring:       Lattice description
         refpts1:    Selection of the 1st family
         refpts2:    Selection of the 2nd family
-        newval:     New tunes
+        newval:     New tunes, in case an non-zero integer part
+                    is provided, fit_integer is set to True
         tol:        Tolerance for the matching; Default: 1.0e-12
         dp:         Momentum deviation. Default: 0
         niter:      Maximum number of iterations. Default 3
-    
+        fit_integer: bool (default=False), use integer tune
```

Comment:
`fit_integer` is not in the argument list. It should be in a "Keyword Arguments:" section.